workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -65,24 +71,126 @@ private static void LaunchMobileClient()
             }
         }
 
-        private static bool TryGetApkPath(string rootPath, out string apkPath)
+        [MenuItem(MenuLaunchiOSDevice, false, 74)]
+        private static void LaunchiOSDeviceClient()
         {
-            foreach (var file in Directory.GetFiles(rootPath, "*.apk", SearchOption.AllDirectories))
+            try
             {
-                apkPath = file;
-                return true;
+                if (IDeviceInstallerBinary == string.Empty)
+                {
+                    Debug.LogError("Could not find ideviceinstaller tool. Please ensure it is installed.");
+                    return;
+                }
+
+                if (IDeviceDebugBinary == string.Empty)
+                {
+                    Debug.LogError("Could not find idevicedebug tool. Please ensure libimobiledevice is installed.");
+                    return;
+                }
+
+                EditorUtility.DisplayProgressBar("Launching iOS Device Client", "Installing archive", 0.3f);
+
+                // Find archive to install
+                var ipaPath = TryGetArchivePath(AbsoluteAppBuildPath, "*.ipa");
+                if (ipaPath == string.Empty)
+                {
+                    Debug.LogError($"Could not find a built out iOS archive in \"{AbsoluteAppBuildPath}\" to launch.");
+                    return;
+                }
+
+                if (RedirectedProcess.Run(IDeviceInstallerBinary, "-i", ipaPath) != 0)
+                {
+                    Debug.LogError("Error while installing archive to the device. Please check the log for details about the error.");
+                    return;
+                }
+
+                EditorUtility.DisplayProgressBar("Launching iOS Device Client", "Launching Client", 0.9f);
+
+                // Get chosen android package id and launch
+                var bundleId = PlayerSettings.GetApplicationIdentifier(BuildTargetGroup.Android);
+
+                // Optional arguments to be passed, same as standalone
+                // Use this to pass through the local ip to connect to
+                var runtimeIp = GdkToolsConfiguration.GetOrCreateInstance().RuntimeIp;
+                var arguments = new StringBuilder();
+                if (!string.IsNullOrEmpty(runtimeIp))
+                {
+                    arguments.Append($"-e SPATIALOS_ARGUMENTS=\"+{RuntimeConfigNames.ReceptionistHost} {runtimeIp}\"");
+                }
+
+                StartiOSDeviceProcess(arguments.ToString(), bundleId);
+
+                EditorUtility.DisplayProgressBar("Launching iOS Device Client", "Done", 1.0f);
             }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        private static string TryGetArchivePath(string rootPath, string archivePattern)
+        {
+            foreach (var file in Directory.GetFiles(rootPath, archivePattern, SearchOption.AllDirectories))
```

Comment:
This could just be replaced with `return Directory.GetFiles(rootPath, archivePattern, SearchOption.AllDirectories).FirstOrDefault();`
What happens if there are multiple archives in the `rootPath`?

Reply:
Will just pick first of them, as we assume unity project only contains one Android/iOS archive. Ideally we would want to allow people to select archive to launch, but I think this functionality should suffice for now